libs/core/src/main/java/org/elasticsearch/core/IOUtils.java

```diff
@@ -95,50 +113,38 @@ public static void close(final Exception e, final Closeable... objects) throws I
      * @param objects objects to close
      */
     public static void close(final Iterable<? extends Closeable> objects) throws IOException {
-        close(null, objects);
-    }
-
-    /**
-     * Closes all given {@link Closeable}s. If a non-null exception is passed in, or closing a
-     * stream causes an exception, throws the exception with other {@link RuntimeException} or
-     * {@link IOException} exceptions added as suppressed.
-     *
-     * @param ex existing Exception to add exceptions occurring during close to
-     * @param objects objects to close
-     *
-     * @see #close(Closeable...)
-     */
-    public static void close(final Exception ex, final Iterable<? extends Closeable> objects) throws IOException {
-        Exception firstException = ex;
+        Exception firstException = null;
         for (final Closeable object : objects) {
             try {
                 close(object);
             } catch (final IOException | RuntimeException e) {
-                if (firstException == null) {
-                    firstException = e;
-                } else {
-                    firstException.addSuppressed(e);
-                }
+                firstException = addOrSuppress(firstException, e);
             }
         }
 
         if (firstException != null) {
-            if (firstException instanceof IOException) {
-                throw (IOException) firstException;
-            } else {
-                // since we only assigned an IOException or a RuntimeException to ex above, in this case ex must be a RuntimeException
-                throw (RuntimeException) firstException;
-            }
+            throwRuntimeOrIOException(firstException);
         }
     }
 
+    private static Exception addOrSuppress(Exception firstException, Exception e) {
```

Comment:
The naming here is confusing. Perhaps call it `maybeSuppress`?

Reply:
I chose this because we have the exact same method in the server package's exception helpers and was going to dry this up in a follow-up. But yea it is a confusing name I agree, we can change it on the utility method in server in that follow up :)